File: listings/cl-existence-oriented/next-set-item.cxx
#include <iostream>
#include <set>

void unwrap_print_set_element( const std::set<int>& elements ) {
    const auto wrapped_element = elements.cbegin( );
    if ( wrapped_element != elements.cend( ) ) {
        std::cerr << "set element: " << *wrapped_element << std::endl;
    }
}

int main ( ) {
    const std::set<int> nothing;
    const std::set<int> things { 42 };

    unwrap_print_set_element( things );
    unwrap_print_set_element( nothing );
}

File: listings/cl-existence-oriented/next-set-item.rs
use std::collections::HashSet;
use std::io::{stderr, Write};

fn main() {
    let nothing = HashSet::new();
    let mut things = HashSet::new();
    things.insert(42);
    unwrap_print_set_element(things);
    unwrap_print_set_element(nothing);
}

fn unwrap_print_set_element(elements: HashSet<i32>) {
    let mut iterator = elements.iter();
    if let Some(element) = iterator.next() {
        writeln!(stderr(), "set element: {}", element).ok();
    }
}
